timesketch/lib/analyzers/feature_plugins/winevt_features.py

```diff
@@ -0,0 +1,204 @@
+
+        if not isinstance(config.get("event_version"), int):
+            raise ValueError("event_version is missing or has invalid value")
+
+    def extract_features(self, name: str, config: dict) -> str:
+        """Extracts features from events.
+
+        Args:
+            name (str): Features extraction name.
+            config (dict): A dict that contains the configuration fo the
+                feature extraction.
+
+        Returns:
+            str: Returns summary of the feature extraction for `feature_name`.
+        """
+        self.validate_feature_config(name, config)
+
+        source_name = config.get("source_name", [""])[0]
+        provider_identifier = config.get("provider_identifier", [""])[0]
+
+        # If event_identifier and event_version does not exist or invalid value type,
+        # it will throw exception.
+        #
+        # Note: Each feature extraction is run as a new celery task so it only affects
+        # specific feature extraction.
+        event_identifier = int(config.get("event_identifier"))
+        event_version = int(config.get("event_version"))
+
+        mappings = config.get("mapping")
+
+        if not mappings:
+            raise ValueError(f"mapping value for {name} is empty")
+
+        # Building search query - Use source_name or provider_identifier and not both.
+        query = ""
+        if source_name:
+            query = f"source_name:{source_name}"
+        elif provider_identifier:
+            query = f"provider_identifier:{provider_identifier}"
+
+        query = (
+            f"{query} AND event_identifier: {event_identifier}"
+            f" AND event_version: {event_version}"
+        )
+
+        events = self.analyzer_object.event_stream(
+            query_string=query, return_fields=self.EVENT_FIELDS
+        )
+        event_counter = 0
+
+        for event in events:
+            attributes = {}
+            strings = event.source.get("strings", None)
+
+            if not strings or not isinstance(strings, list):
+                logger.debug(
+                    "Missing or invalid strings field in the event. Skipping the"
+                    " event %s.",
+                    event.event_id,
+                )
+                continue
+
+            for mapping in mappings:
+                attribute_name = mapping.get("name")
+                if not attribute_name:
+                    logger.debug(
+                        "Skipping event %s. Missing attribute name.", event.event_id
+                    )
+                    continue
+
+                try:
+                    string_index = int(mapping.get("string_index"))
+                except (TypeError, ValueError) as e:
+                    logger.debug(
+                        "Skipping event %s. Missing string_index or invalid value. %s",
+                        event.event_id,
+                        str(e),
+                    )
+                    continue
```

Comment:
These checks should be done in the `validate_feature_config` function. If a feature definition is not following the defined schema, we should not map any of the values.

```suggestion
                string_index = int(mapping.get("string_index"))
```